### ISSUE TITLE
Implement realistic income calculations for agriculture

### ISSUE THREAD

**Issue by jakobkolb** [jakobkolb/MayaSim#2]
So far, per capita income from agriculture is calculated via the mean of the agricultural productivity of the cropped cells of a settlement divided by the number of its inhabitants.

``` python
    def get_crop_income(self,bca):
        # agricultural benefit of cropping
        for city in np.where(self.population!=0)[0]:
            crops = bca[self.cropped_cells[city][0],self.cropped_cells[city][1]]
            self.crop_yield[city] = np.nanmean(crops[crops>0])            
        self.crop_yield[np.isnan(self.crop_yield)] = 0
        self.crop_yield[self.population==0] = 0
        return self.crop_yield
```

this does not make sense. I would say, one should rather sum over the agricultural production of the cropped cells of a settlement.

**Comment by jakobkolb** [comment]
Now, one can choose between average and sum over the agricultural productivity of cropped cells of a settlement.

``` python
    def get_crop_income(self,bca):
        # agricultural benefit of cropping
        for city in np.where(self.population!=0)[0]:
            crops = bca[self.cropped_cells[city][0],self.cropped_cells[city][1]]
            if self.crop_income_mode == "mean":
                self.crop_yield[city] = np.nanmean(crops[crops>0])
            elif self.crop_income_mode == "sum":
                self.crop_yield[city] = np.nansum(crops[crops>0])            
        self.crop_yield[np.isnan(self.crop_yield)] = 0
        self.crop_yield[self.population==0] = 0
        return self.crop_yield
```

The question is, whether the weighting of different sources of income in the _national accounts_ is out of balance now and needs some fine tuning. 

``` python
    def get_real_income_pc(self):
        ### combine agricultural, ecosystem service and trade benefit          
        self.real_income_pc = (r_bca * self.crop_yield 
                                         + r_es * self.eco_benefit 
                                         + r_trade * self.trade_income)/ self.population         
        return self.real_income_pc
```

Experiments concerning this question are ongoing.

**Comment by jakobkolb** [comment]
First experiments show that with 'sum' income calculation method, the income from agriculture is way to high, compared to income from other sources. 

Income from different sources with 'mean' agriculture income:
![mean_income_panel](https://cloud.githubusercontent.com/assets/2690855/14849714/dc08851e-0c74-11e6-8bc8-9cbbd8514341.png)

Income from different sources with 'sum' agriculture income:
![sum_income_panel](https://cloud.githubusercontent.com/assets/2690855/14849715/dc09a502-0c74-11e6-9685-efaebe44599d.png)

To get realistic results, agriculture income has to be rescaled by a approximately a factor 10^-1.

**Comment by jakobkolb** [comment]
Updated the calculation of real income such that trade-, agriculture-, and ecosystems services income is rescaled where it is calculated:

``` python
    def get_crop_income(self,bca):
        # agricultural benefit of cropping
        for city in self.populated_cities:
            crops = bca[self.cropped_cells[city][0],self.cropped_cells[city][1]]
            if self.crop_income_mode == "mean":
                self.crop_yield[city] = r_bca_mean*np.nanmean(crops[crops>0])
            elif self.crop_income_mode == "sum":
                self.crop_yield[city] = r_bca_sum*np.nansum(crops[crops>0])            
        self.crop_yield = [0 if np.isnan(self.crop_yield[index]) \
                else self.crop_yield[index] for index in range(len(self.crop_yield))]
        return
```

and at total real income calculation the different income types are only summed up.

``` python
    def get_real_income_pc(self):
        ### combine agricultural, ecosystem service and trade benefit     
        self.real_income_pc = [(self.crop_yield[index] \
                + self.eco_benefit[index] \
                + self.trade_income[index]) \
                /self.population[index] \
                if value > 0 else 0 \
                for index, value in enumerate(self.population)]
        return
```

Therefore, the output data has the correct scales and can be plotted without any further modification (that was hardcoded before, as I realized today..)

The updated code is currently running on the cluster and first results should be ready tomorrow.

**Comment by jakobkolb** [comment]
For the parameter scans of r_es and r_ag, the experiments fail due to 
* killing cities that are out of bounds of the city array, 
* concatenating arrays of occupied cells where there are none.

**Comment by jakobkolb** [comment]
Bugs fixed and calibration results committed in  7bbef1fd9da80ba52983f123ba1e136e59453747
